stacks-core/src/wallet.rs

```diff
@@ -0,0 +1,295 @@
+/*!
+Exposes tools to create and manage Stacks credentials.
+*/
+
+use std::str::FromStr;
+
+use bdk::bitcoin::secp256k1::Secp256k1;
+use bdk::bitcoin::util::bip32::{DerivationPath, ExtendedPrivKey};
+use bdk::bitcoin::{
+    Address as BitcoinAddress, AddressType as BitcoinAddressType, Network as BitcoinNetwork,
+};
+use bdk::keys::bip39::Mnemonic;
+use rand::random;
+use serde::{Deserialize, Serialize};
+
+use crate::address::{AddressVersion, StacksAddress};
+use crate::crypto::wif::WIF;
+use crate::StacksError;
+use crate::{
+    crypto::{PrivateKey, PublicKey},
+    Network, StacksResult,
+};
+
+/// Computes Stacks derivation paths
+pub fn stacks_derivation_path(index: u32) -> StacksResult<DerivationPath> {
+    Ok(DerivationPath::from_str(&format!(
+        "m/44'/5757'/0'/0/{}",
+        index
+    ))?)
+}
+
+/// Computes Bitcoin derivation paths
+pub fn bitcoin_derivation_path(
+    network: BitcoinNetwork,
+    kind: BitcoinAddressType,
+    index: u32,
+) -> StacksResult<DerivationPath> {
+    let mut path = "m/".to_string();
+
+    match kind {
+        BitcoinAddressType::P2pkh => path.push_str("44'/"),
+        BitcoinAddressType::P2wpkh => path.push_str("84'/"),
```

Comment:
Do you want to add m/49' ? https://github.com/bitcoin/bips/blob/master/bip-0049.mediawiki